src/libraries/System.IO.FileSystem.AccessControl/src/System/Security/AccessControl/DirectorySecurity.cs

```diff
@@ -16,7 +16,6 @@ public DirectorySecurity()
         public DirectorySecurity(string name, AccessControlSections includeSections)
             : base(true, name, includeSections, true)
         {
-            Path.GetFullPath(name);
```

Comment:
This validation was not being done on time. By the time we needed a valid path, we ended up throwing a different error deeper in the callstack.

Now we call GetFullPath in the ancestor's constructor, ensuring we throw early in case the path is invalid.

Reply:
I'm not sure I understand. This is a public constructor. Now it will no longer throw the same way for embedded nulls in the path, will it not?

Reply:
Ah, never mind, I see. I misunderstood, you are calling it in the base constructor.